src/main/java/com/instaclick/pentaho/plugin/amqp/AMQPPlugin.java

```diff
@@ -100,6 +162,14 @@ public boolean processRow(StepMetaInterface smi, StepDataInterface sdi) throws K
 
                 setOutputDone();
 
+                //now output rowset closed, so we can wait for our confirmation steps
+                if ( data.activeConfirmation )  {
+                    while (!isStopped() && data.watchedConfirmStep.size() > 0) {    
+                        Thread.sleep(500);

```

Comment:
one of my use-case
1. consumer got 10000 messages, and wait for tags for ACK from ACK step
2. consumer supplies messages to Table Output step which inserts in batch by 2000 messages.
3. Table Output step sends data to ACK dummy step by portion of 2000 messages after all of them inserted into DB
4. Consumer sees those batch of data on ACK input by RowListener events, and can immidiately ack those messages from queue. at this time we have 100% assured that data inserted into DB